src/main/kotlin/hr/foi/pop/backend/response_bodies/ActivateUserResponseBody.kt

```diff
@@ -0,0 +1,27 @@
+package hr.foi.pop.backend.response_bodies
+
+import hr.foi.pop.backend.models.user.User
+
+class ActivateUserResponseBody(
+    val id: Int,
+    val firstName: String,
+    val lastName: String,
+    val username: String,
+    val email: String,
+    val role: String,
+    val activated: Boolean
+) {
+    companion object {
+        fun mapResponse(user: User): ActivateUserResponseBody {
+            return ActivateUserResponseBody(
+                user.id,
+                user.firstName,
+                user.lastName,
+                user.username,
+                user.email,
+                user.role.name,
+                user.isAccepted
+            )
+        }
+    }
+}
```

Comment:
You don't need this class at all! You could just send a User object directly as a parameter to SuccessResponse, as in:
```kotlin
/// In UserController...
SuccessResponse(
    "User '${user.username}' ${activateOrDeactivate.label}",
    user // <- just the pure DTO object!
)
```

And you could argue that, if the UserDTO changes, then the response is going to change. I say - sure, when that happens, a few of your tests are (hopefully) going to fail and then you can think about mapping the UserDTO to a special response just for this use-case.

Reply:
I consider to do it like that. But then I get more data than needed. Response will also have additional information such as stores, event, balance and so on.
Still I go back to UserDTO or should I leave this type of response?

Reply:
Maybe map additional information as null before sending to client?

Reply:
Oh, well, look: as far as the "more data than needed" goes, I find that to be a good thing. Facebook had a problem with that so they invented GraphQL. We're not going to have millions of users so I say getting more data than needed is just fine, especially for future-proofing the API for any other application that might use our endpoints.

I would go back to the simplest solution possible, which I find to be the one I suggested. It involves 1 line over an entire new 30-line class.

Mapping additional information to null sounds like a dangerous idea that could drastically go wrong one day when this API in particular expands, in any direction. Let's keep it short and simple. 😃